### ISSUE TITLE
[BUG]: Simpit crash when using targets

### ISSUE THREAD

**Issue by TheDicko** [LRTNZ/KerbalSimpitRevamped#8]
# Description of the bug
Simpit crash surrounding targets (Note this happens regardless of whether target channel is subscribed or not).

### What feature are you trying to use that is in the documentation, that is not behaving as the documentation says it should/will do?  

When setting a target, and then going to the tracking station (without un setting the target, and then when you select another vessel to fly, it crashes the plugin.

I am displaying data to tft screen specifically, but it seems this is a universal bug rather than related to displays. 

### What have you done to ensure the issue is a bug? Have you tried checking different combinations of button presses for example, to ensure that it is indeed not your controllers code?  
<br>

### How repeatable is it? Eg "After I press stage 100 times, the mod will not take any more inputs from my controller"  
Crashes every time with any vessels.
<br>

## To Reproduce
1. Please provide the code of your controller, to help with diagnosing the issue.
2. Please thoroughly explain the most repeatable way that you have found to get the issue you are experiencing to occur.
3. Please include any other details that you have that could be relevant to help with solving the problem.

1. My code is 99% specific to the tft screen i am using. I beleive it will be unuseful, however i can send it if it will be needed.
2. Commenting out most of the code in my arduino sketch, and testing as i described earlier.
3. Nothing to add (i PM'd you most of the details earlier as you know).

## Expected behaviour
Not to crash 

### A clear and concise description of what you expected to happen.

The Game seems to keep a reference of which vessel is targeting another object, ragardless of whether it is currently being flown by the user. Currently after looking at the simpit source code, it seems the plugin assumes a vessel is also tageting the same object as the previous vessel, and therefore it crashes. 

### What is it in the documentation that leads you to believe the behaviour you are experiencing is unintended. Eg: Is something meant to be returning a specific value, and is never doing that?

Nothing in the documentation, just my (hopefully correct) understanding of how the game works and the mod

## Additional context
- Please attach your code for the controller you are using/trying to build.
- Any log files you may have

Here is the log when it crashes.

KerbalSimpit.Providers.KerbalSimpitTargetProvider.TargetProvider () [0x00028] in <f706074555ae4f73b5b5c4eae7443a4c>:0
at KerbalSimpit.KSPit+<>c__DisplayClass14_0.<EventWorker>b__0 () [0x00061] in <f706074555ae4f73b5b5c4eae7443a4c>:0
at KerbalSimpit.KSPit.EventWorker () [0x0003b] in <f706074555ae4f73b5b5c4eae7443a4c>:0
at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00014] in <ad04dee02e7e4a85a1299c7ee81c79f6>:0
at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00071] in <ad04dee02e7e4a85a1299c7ee81c79f6>:0
at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <ad04dee02e7e4a85a1299c7ee81c79f6>:0
at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x0002b] in <ad04dee02e7e4a85a1299c7ee81c79f6>:0
at System.Threading.ThreadHelper.ThreadStart () [0x00008] in <ad04dee02e7e4a85a1299c7ee81c79f6>:0
UnityEngine.DebugLogHandler:Internal_LogException(Exception, Object)
UnityEngine.DebugLogHandler:LogException(Exception, Object)
ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
UnityEngine.Logger:LogException(Exception, Object)
UnityEngine.Debug:LogException(Exception)
UnityEngine.UnhandledExceptionHandler:<RegisterUECatcher>m__0(Object, UnhandledExceptionEventArgs)

**Comment by PBechon** [comment]
I managed to reproduce it without even having to go to the tracking station. 

I launch a new game and load a rocket on the launching pad.
I launch the rocket : no problem and my controller works;
I revert the launch
I set the Mun as a target and launch the rocket : the plugin crashed and stop sending data to arduino.

Tested with the KerbalSimpitAltitudeTrigger example on Arduino Uno.

Here is the extract of my KSP.log

> [EXC 21:19:54.849] NullReferenceException: Object reference not set to an instance of an object
> 	KerbalSimpit.Providers.KerbalSimpitTargetProvider.TargetProvider () (at <f706074555ae4f73b5b5c4eae7443a4c>:0)
> 	KerbalSimpit.KSPit+<>c__DisplayClass14_0.<EventWorker>b__0 () (at <f706074555ae4f73b5b5c4eae7443a4c>:0)
> 	KerbalSimpit.KSPit.EventWorker () (at <f706074555ae4f73b5b5c4eae7443a4c>:0)
> 	System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) (at <ad04dee02e7e4a85a1299c7ee81c79f6>:0)
> 	System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) (at <ad04dee02e7e4a85a1299c7ee81c79f6>:0)
> 	System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) (at <ad04dee02e7e4a85a1299c7ee81c79f6>:0)
> 	System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) (at <ad04dee02e7e4a85a1299c7ee81c79f6>:0)
> 	System.Threading.ThreadHelper.ThreadStart () (at <ad04dee02e7e4a85a1299c7ee81c79f6>:0)
> 	UnityEngine.DebugLogHandler:LogException(Exception, Object)
> 	ModuleManager.UnityLogHandle.InterceptLogHandler:LogException(Exception, Object)
> 	UnityEngine.UnhandledExceptionHandler:<RegisterUECatcher>m__0(Object, UnhandledExceptionEventArgs)

**Comment by LRTNZ** [comment]
Ok, that's interesting. Given that it is evidently an easily replicable bug, I will take a good look into it.

**Comment by PBechon** [comment]
I tried to pinpoint the issue and I think I have a fix (see #9 ).

This is my first interaction with KSP codebase, so I might have missed something. But I think the getVessel() method that was called on all targets returned a null pointer when the target was not a Vessel (in my case, it was a Celestial body).
@TheDicko what was your target ?

**Comment by TheDicko** [comment]
@PBechon I think it was too a celestial body, but i may be wrong. Regardless i'll test the most recent version.

**Comment by LRTNZ** [comment]
> 
> 
> I tried to pinpoint the issue and I think I have a fix (see #9 ).
> 
> This is my first interaction with KSP codebase, so I might have missed something. But I think the getVessel() method that was called on all targets returned a null pointer when the target was not a Vessel (in my case, it was a Celestial body).
> @TheDicko what was your target ?

That does make sense, to directly look for the target instead of calling the getVessel() on all targets. As if the target is a Celestial body, ofc it will not have the getVessel() method.

I will test this a bit later, and merge it in! How have you been compiling/installing to test? Using Visual Studio?

**Comment by PBechon** [comment]
Yes I did the quick and dirty thing ... I cloned the repository, imported it as a new project in VisualStudio, added the reference to the Unity DLL in KSP_x64_Data/Managed and hit compile. After I copy the DLL in the GameData repository (replacing the original KerbalSimpit DLL).

The drawback is that I did not found a way to debug while KSP is running. So the process to test is tedious : I add a Debug statement, compile and copy the DLL, launch KSP (it takes ages ...), launch the game, realize I did not print the right thing, close the game and do it again ...
I have even uninstalled all my graphics mod to reduce KSP load time but it is still too much.
